src/lib.rs

```diff
@@ -51,46 +51,41 @@ impl<T> Part for T where T: Encodable + Decodable {
     }
 }
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, RustcDecodable, RustcEncodable)]
 /// A basic JWT header part, the alg is automatically filled for use
-/// It's missing things like the kid but that's for later
+/// and the algorithm defaults to HS256
 pub struct Header {
-    typ: &'static str,
+    typ: String,
     alg: Algorithm,
+    jku: Option<String>,
+    kid: Option<String>,
+    x5u: Option<String>,
+    x5t: Option<String>
 }
 
 impl Header {
     pub fn new(algorithm: Algorithm) -> Header {
         Header {
-            typ: "JWT",
+            typ: "JWT".to_owned(),
             alg: algorithm,
+            jku: None,
+            kid: None,
+            x5u: None,
+            x5t: None
         }
     }
 }
 
-impl Part for Header {
-    type Encoded = &'static str;
-
-    fn from_base64<B: AsRef<[u8]>>(encoded: B) -> Result<Self, Error> where Self: Sized {
-        let algoritm = match encoded.as_ref() {
-            b"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9" => { Algorithm::HS256 },
-            b"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzM4NCJ9" => { Algorithm::HS384 },
-            b"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9" => { Algorithm::HS512 },
-            _ => return Err(Error::InvalidToken)
-        };
-
-        Ok(Header::new(algoritm))
+impl Default for Header {
+    fn default() -> Header {
+        Header::new(Algorithm::HS256)
     }
+}
 
-    fn to_base64(&self) -> Result<Self::Encoded, Error> {
-        let encoded = match self.alg {
-            Algorithm::HS256 => { "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9" },
-            Algorithm::HS384 => { "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzM4NCJ9" },
-            Algorithm::HS512 => { "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9" },
-        };
-
-        Ok(encoded)
-    }
+#[derive(Debug)]
```

Comment:
This blurs the line between header and claims a bit too much for my taste. I've looked at tuple but it seems like it would be a downgrade compared to a struct in this case